flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/LocalStreamEnvironment.java

```diff
@@ -91,7 +91,11 @@ public JobExecutionResult execute(String jobName) throws Exception {
 		configuration.addAll(jobGraph.getJobConfiguration());
 
 		configuration.setLong(ConfigConstants.TASK_MANAGER_MEMORY_SIZE_KEY, -1L);
-		configuration.setInteger(ConfigConstants.TASK_MANAGER_NUM_TASK_SLOTS, getParallelism());
+
+		int parallelism = getParallelism() == defaultLocalParallelism ?
+				defaultLocalParallelism : jobGraph.getMaximumParallelism();
```

Comment:
This currently means that in the default case, the slots do not respect the max parallel operator, correct?